### PR TITLE
Adding layout structure, CSS, and API Reference Controllers

### PR DESCRIPTION
**Pull request by matthewjohnston4**

### CODE REVIEW

File: src/site/WebController.php [hunk 1]
```diff
@@ -18,6 +18,15 @@ protected function getRequiredStringParam(string $key): string {
     return $this->parameters[$key];
   }
 
+  protected function getStringParam(
+    string $key, string $default = ""
+  ): string {
```
Comment (fredemmott) [inline src/site/WebController.php]: would prefer for this to return ?string and possibly rename to getOptionalStringParam

**Comment by fredemmott** [comment]
getSideNav() looks like it should move into a superclass, and it and related functions should be async

**Comment by fredemmott** [comment]
Closing - can this be re-submitted as 3 separate parts?

File: src/site/APIListController.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?hh // strict
+
+use HHVM\UserDocumentation\APIIndex;
+
+enum APIProduct: string as string {
+  HACK = 'hack';
+}
+
+final class APIListController extends WebPageController {
+  protected async function getTitle(): Awaitable<string> {
```
Comment (SiebelsTim) [inline src/site/APIListController.php]: This doesn't always return

**Pull request closed without merging**